src/modules/linux-micro/kmod/kmod.c

```diff
@@ -0,0 +1,474 @@
+
+    r = readdir_r(dir, &ent, &entbuf);
+    while (r == 0 && entbuf) {
+        char *s, *dot = strrchr(ent.d_name, '.');
+
+        if (!dot || !streq(dot, ".conf")) {
+            SOL_DBG("File name does not end in '.conf', ignoring: %s",
+                ent.d_name);
+            goto next;
+        }
+
+        s = strdup(ent.d_name);
+        if (!s) {
+            SOL_WRN("Could not allocate memory to hold dir entry: %s, skipping.",
+                ent.d_name);
+            goto next;
+        }
+
+        sol_ptr_vector_insert_sorted(vec, s, kmod_conf_cmp);
+
+next:
+        r = readdir_r(dir, &ent, &entbuf);
+    }
+}
+
+static void
+uevent_read_msg(struct kmod_data *kdata, char *msg, int len)
+{
+    char *modalias = NULL, *action = NULL;
+    int i = 0;
+
+    while (i < len) {
+        char *curr, *del;
+        size_t value_len;
+
+        curr = msg + i;
+        del = strrchr(curr, '=');
+        value_len = del - curr;
+
+        if (del && !strncmp(curr, "MODALIAS", value_len))
+            modalias = strdupa(++del);
+        else if (del && !strncmp(curr, "ACTION", value_len))
+            action = strdupa(++del);
+
+        i += strlen(msg + i) + 1;
+    }
+
+    if ((action && modalias) && !strcmp(action, "add")) {
+        kmod_apply_value(kdata->kmod, modalias);
+    }
+}
+
+static bool
+kmod_uevent_handler(void *data, int fd, unsigned int cond)
+{
+    int len;
+    struct kmod_data *kdata = data;
+    char buffer[512];
+
+    len = recv(kdata->uevent.fd, buffer, sizeof(buffer), MSG_WAITALL);
+    if (len == -1) {
+        SOL_ERR("Could not read netlink socket. %s", sol_util_strerrora(errno));
+        return false;
+    } else {
+        uevent_read_msg(kdata, buffer, len);
+    }
+
+    return true;
+}
+
+static int
+kmod_uevent_start(struct kmod_data *kdata)
+{
+    struct sockaddr_nl nls;
+
+    memset(&nls, 0, sizeof(struct sockaddr_nl));
+    nls.nl_family = AF_NETLINK;
+    nls.nl_pid = getpid();
+    nls.nl_groups = -1;
+
+    kdata->uevent.fd = socket(PF_NETLINK, SOCK_DGRAM, NETLINK_KOBJECT_UEVENT);
+    if (kdata->uevent.fd == -1) {
+        SOL_ERR("Could not open uevent netlink socket.");
+        return -errno;
+    }
+
+    if (bind(kdata->uevent.fd, (void *)&nls, sizeof(struct sockaddr_nl))) {
+        SOL_ERR("Could not bind to uevent socket.");
+        close(kdata->uevent.fd);
+        kdata->uevent.fd = 0;
+        return -errno;
+    }
+
+    kdata->uevent.watch = sol_fd_add(kdata->uevent.fd, SOL_FD_FLAGS_IN,
+        kmod_uevent_handler, kdata);
+    return 0;
+}
+
+static void
+kmod_coldplug_find_devices(struct kmod_data *kdata, DIR *dir)
+{
+    struct dirent ent, *entbuf = NULL;
+    int r, dfd;
+
+    dfd = dirfd(dir);
+    if (dfd < 0) {
+        SOL_DBG("Could not obtain directory file descriptor");
+        goto next;
+    }
+
+    r = readdir_r(dir, &ent, &entbuf);
+    while (r == 0 && entbuf) {
+        struct stat st;
+
+        if (!strcmp(ent.d_name, ".") || !strcmp(ent.d_name, "..")) {
```

Comment:
Yes, but this is skipping only `.` and `..`. Hidden files will still be processed.